pkg/certificate/chain/api.go

```diff
@@ -0,0 +1,90 @@
+package chain
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"time"
+
+	"github.com/qinqon/kube-admission-webhook/pkg/certificate/triple"
+)
+
+// CertificateIssue contains details about an issued certificate, including the
+// private key and named CA certificates known to verify the issued certificate.
+type CertificateIssue struct {
+	Name      string
+	IPs       []string
+	Hostnames []string
+	KeyPEM    []byte
+	CertPEM   []byte
+	CACertPEM map[string][]byte
```

Comment:
I have to go back to this struct, why do we need multiple CACerts per caBundle ? all the CABundle will share the same content I think we don't need a map here.

Reply:
Ensuring that all the ca bundles share the some content requires specific logic that was not there in the previous implementation either (where it would only append, not replace, to the ca bundle). That logic would be now more relevant & complicated given that we support multiple webhook configs with possibly different lifetimes. So in this case I think it is more useful & simple to have a faithful representation of the data structure, which allows for further improvements in the future, let's say for example if we would like to handle different webhooks at different times as well.

Reply:
We don't want generic mapping this is an opinionated cert-rotation lib desgined for addmission webhooks so referencing CABundleName is ok.

Also maps naming maps with ValueByKey makes clear is a map while KeydValue could be a slice.

Reply:
I understand that but that completely breaks a design decision I made when refactoring that was abstracting the certificate chain rotation. So to be sensible to this approach, is there no alternative possibility to clarify, either by more clear generic naming or more doc what this is, as a generalized concept? Because otherwise we are up for renaming everything from now on, reversing all considerations that I had while working on this.

Reply:
Ack, let's keep this generic then, but let's use "ByName" since it's a known golang pattern to name maps this way.

Reply:
great!